processor/intracesampler/fnvhasher.go

```diff
@@ -0,0 +1,37 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package intracesampler // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/intracesamplerprocessor"
+
+import (
+	"hash/fnv"
+)
+
+// computeHash creates a hash using the FNV-1a algorithm
+func computeHash(b []byte, seed []byte) uint32 {
+	hash := fnv.New32a()
+	// the implementation fnv.Write() does not return an error, see hash/fnv/fnv.go
+	_, _ = hash.Write(seed)
+	_, _ = hash.Write(b)
+	return hash.Sum32()
+}
+
+// i32tob converts a seed to a byte array to be used as part of fnv.Write()
+func i32tob(val uint32) []byte {
+	r := make([]byte, 4)
```

Comment:
Make sure you define magic numbers somewhere in case anyone needs to change at least understand what they are intended for.

Reply:
This too is copied as-is from the existing [probabilistic sampler processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/85198a60c315d75fcf74b6b84a6504a6e699e00a/processor/probabilisticsamplerprocessor/fnvhasher.go#L22). 

I prefer if this improvement be done in a dedicated PR for both places.

Reply:
That is also fair, if you can use a perma link as to where this code is referenced, it would help those wanting to figure out the reasoning for this code and how it came to existence. ( Bitshifting is a pain to get right the first time in my experience.)

Reply:
added relevant comment in code